src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -549,6 +549,25 @@ generate_treespec (GHashTable *packages,
   return g_steal_pointer (&ret);
 }
 
+static gboolean
+sack_has_subject (DnfSack    *sack,
```

Comment:
Ah, nice.  This approach seems like it's going to solve a bunch of problems indeed.